frontend/src/utils/video.ts

```diff
@@ -60,140 +64,40 @@ export function idFromUid(uid: string): string {
-  }
-
-  const newUid = await retryRandomPick(5, uid, uidOther, newSuggestions);
-  return newUid;
-}
-
-export async function getUidFromComparedList(
-  uid: string | null,
-  uidOther: string | null,
-  exclude?: string[]
-): Promise<string | null> {
-  const contributorRatings = await UsersService.usersMeContributorRatingsList({
-    pollName: YOUTUBE_POLL_NAME,
-    limit: 99,
-    offset: 0,
-  });
-
-  let newSuggestions =
-    contributorRatings.results?.map((rating) => rating.entity.uid) || [];
-
-  if (exclude) {
-    newSuggestions = newSuggestions.filter((uid) => !exclude.includes(uid));
-  }
-
-  const newUid = retryRandomPick(5, uid, uidOther, newSuggestions);
-  return newUid;
-}
-
-/**
- * This helper function returns a UID following the strategy:
- *
- *  1. Uniformily random from rate_later list (75% chance)
- *  2. Uniformily random from already rated videos (20% chance)
- *  3. Uniformily random from Tournesol's top 100 videos (5% chance)
- *  If option 1 is selected and fails, option 2 will be tried
- *  If option 2 is selected and fails, option 3 will be tried
- *
- * The exlude parameter allows to exclude a list of UIDs before the random
- * selection.
- */
 export async function getUidForComparison(
   uid: string | null,
   uidOther: string | null,
-  exclude?: string[]
+  poll: string
 ): Promise<string | null> {
-  const x = Math.random();
-
-  let newUid;
-
-  if (x < 0.75) {
-    newUid = await getUidFromRateLaterList(uid, uidOther, exclude);
-    if (newUid) return newUid;
-  }
+  const exclude: string[] = [];
 
-  if (x < 0.95) {
-    const newUid = await getUidFromComparedList(uid, uidOther, exclude);
-    if (newUid) return newUid;
-  }
-
-  const videoResult = await PollsService.pollsRecommendationsList({
-    name: 'videos',
-    limit: 100,
-    // Increase the diversity of recommended videos. Changing the `offset`
-    // allows us to not increase the range of suggested videos, without
-    // increasing size of the downloaded JSON. We may want to adapt/remove
-    // this `offset` the day we will filter the results according to the user
-    // preferred languages.
-    offset: getPseudoRandomInt(10) * 10,
+  [uid, uidOther].forEach((item) => {
+    if (item) {
+      exclude.push(item);
+    }
   });
 
-  let newSuggestions = (videoResult?.results || []).map((v) => v.entity.uid);
-
-  if (exclude) {
-    newSuggestions = newSuggestions.filter((uid) => !exclude.includes(uid));
+  if (isAutoSuggestionsEmpty(poll)) {
+    const suggestions = await UsersService.usersMeSuggestionsTocompareRetrieve({
+      pollName: poll,
+    });
+
+    if (suggestions['count'] > 0) {
+      fillAutoSuggestions(
+        poll,
+        suggestions['results'].map(
+          (item: { entity: RelatedEntity }) => item.entity.uid
+        ),
+        exclude
+      );
+    }
   }
 
-  newUid = await retryRandomPick(5, uid, uidOther, newSuggestions);
+  if (isAutoSuggestionsEmpty(poll)) {
+    return null;
```

Comment:
Will this case ever happen? What would/could be the cause?
How is `null` handled by the video comparison card component?

Reply:
It's very unlikely but it can happen after a new installation without any data, like the first deployment ever of a new Tournesol instance, or a local set up without the `dev-env`.

I think I'm gonna remove this line because the following call  to `autoSuggestionsRandom` already checks if the suggestion array is null.

> How is `null` handled by the video comparison card component?

When the value null is returned, like before, a click on the Auto button does nothing visually. There is no feedback and it's a bit surprising, users may think the Auto button is not working.

Even if this is very situational, it still can happen if the API is down. A task could be to add a visual hint, like a `notistack` message "Sorry, no element found.".